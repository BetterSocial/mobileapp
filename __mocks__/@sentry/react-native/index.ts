export const addBreadcrumb = jest.fn();
