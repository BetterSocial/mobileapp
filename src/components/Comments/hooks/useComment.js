import React from 'react';
import {iVoteComment, voteComment, voteCommentV2} from '../../../service/vote';

const useComment = ({comment, findCommentAndUpdate, level, updateVote}) => {
  const [totalVote, setTotalVote] = React.useState(
    comment.data?.count_upvote - comment.data?.count_downvote
  );
  const [statusVote, setStatusVote] = React.useState('none');
  const onUpVote = async () => {
    if (statusVote === 'upvote') {
      setTotalVote((prevState) => prevState - 1);
      setStatusVote('none');
    }
    if (statusVote === 'downvote') {
      setTotalVote((prevState) => prevState + 2);
      setStatusVote('upvote');
    }
    if (statusVote === 'none') {
      setTotalVote((prevState) => prevState + 1);
      setStatusVote('upvote');
    }
    const dataVote = {
      reaction_id: comment.id,
      vote: 'upvote'
    };
    onVote(dataVote);
  };
  const onDownVote = async () => {
    if (statusVote === 'upvote') {
      setTotalVote((prevState) => prevState - 2);
      setStatusVote('downvote');
    }
    if (statusVote === 'downvote') {
      setTotalVote((prevState) => prevState + 1);
      setStatusVote('none');
    }
    if (statusVote === 'none') {
      setTotalVote((prevState) => prevState - 1);
      setStatusVote('downvote');
    }
    const dataVote = {
      reaction_id: comment.id,
      vote: 'downvote'
    };
    onVote(dataVote);
  };
  const onVote = async (dataVote) => {
    const result = await voteCommentV2(dataVote);
    if (findCommentAndUpdate) {
      findCommentAndUpdate(comment.id, result.data, level);
    }
    if (updateVote) {
      updateVote(result.data, comment, level);
    }
    iVote();
  };

  const iVote = async () => {
    const result = await iVoteComment(comment.id);
    if (result.code === 200) {
      setStatusVote(result.data.action);
    }
  };

  return {
    totalVote,
    setTotalVote,
    statusVote,
    setStatusVote,
    onVote,
    iVote,
    onUpVote,
    onDownVote
  };
};

export default useComment;
