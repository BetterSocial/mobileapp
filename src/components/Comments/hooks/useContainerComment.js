const useContainerComment = () => {
  const isLast = (index, item, comments) =>
    index === comments.length - 1 && (item.children_counts.comment || 0) === 0;

  const isLastInParent = (index, comments) => index === comments.length - 1;

  const hideLeftConnector = (index, comments) => index === comments.length - 1;

  return {
    isLast,
    isLastInParent,
    hideLeftConnector
  };
};

export default useContainerComment;
