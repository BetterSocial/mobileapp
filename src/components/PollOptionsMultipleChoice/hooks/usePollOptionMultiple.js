export const usePollOptionMultiple = ({ 
    item,
    mypoll,
    index,
  // total,
    selectedindex,
    isexpired,
    isalreadypolling = false,
    maxpolls = [],
     onselected = () => {},
    totalVotingUser = 0}) => {

    const counter = item?.counter || 0;
    const optionPercentage = totalVotingUser === 0 ? 0 : ((counter / totalVotingUser) * 100).toFixed(0);

    const isPollDisabled = () => isalreadypolling || isexpired;

    const selected = selectedindex.includes(index);

    const isMyPoll = () => mypoll.reduce((acc, current) => {
      const isCurrentItemMyPoll =
        item?.polling_option_id === current?.polling_option_id;
      return acc || isCurrentItemMyPoll;
    }, false);

    const onOptionsClicked = () => {
    if (isPollDisabled()) {
      return null
    }
    if (selected) {
      const idx = selectedindex.indexOf(index);
      const newSelectedIndex = [...selectedindex];
      newSelectedIndex.splice(idx, 1);
      onselected(newSelectedIndex);
    } else {
      const newSelectedIndex = [...selectedindex];
      newSelectedIndex.push(index);
      onselected(newSelectedIndex);
    }
  };

    const isPollNotEndedAndIsMine = isalreadypolling && isMyPoll() && !isexpired;
    const isMax = maxpolls.includes(item.polling_option_id);

    const handleStyleBar = (percent) => {
      let newPercent = percent
      if(!percent){
        newPercent = 0
      }
      if(percent > 100) {
       newPercent = 100
      }
      return newPercent
    }

    return {
        counter,
        optionPercentage,
        isPollDisabled, 
        selected,
        isMax,
        isMyPoll,
        onOptionsClicked,
        isPollNotEndedAndIsMine,
        handleStyleBar
    }

}



export default usePollOptionMultiple