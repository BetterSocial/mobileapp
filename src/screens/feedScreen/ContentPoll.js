import React from 'react';
import {
  View,
  StyleSheet,
  Text,
  Platform,
  Dimensions,
  Image,
  FlatList,
  TouchableOpacity,
} from 'react-native';

import SeeMore from 'react-native-see-more-inline';
import {ParallaxImage} from 'react-native-snap-carousel';

import Gap from '../../components/Gap';
import PollOptions from '../../components/PollOptions';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const {width: screenWidth} = Dimensions.get('window');

const _renderItem = ({item, index}, parallaxProps) => {
  return (
    <View key={index} style={styles.item}>
      <ParallaxImage
        source={{uri: item}}
        containerStyle={styles.imageContainer}
        style={styles.image}
        parallaxFactor={0.4}
        {...parallaxProps}
      />
    </View>
  );
};

const ContentPoll = ({message, images_url, polls = [], onPress}) => {
  let totalPollCount = polls.reduce((acc, current) => {
    return acc + parseInt(current.counter);
  }, 0);

  return (
    <TouchableOpacity onPress={onPress} style={styles.contentFeed}>
      {images_url !== null ? (
        images_url.length > 0 ? (
          <View
            style={{
              flex: 1,
              paddingBottom: 16,
            }}>
            <SeeMore
              seeLessText={' '}
              numberOfLines={4}
              linkStyle={styles.textContentFeed}>
              {message}
            </SeeMore>
            <Gap height={16} />
            <FlatList
              style={{flex: 1}}
              horizontal={true}
              pagingEnabled={true}
              data={images_url}
              renderItem={({item, index}) => {
                return (
                  <Image
                    source={{uri: item}}
                    style={{flex: 1, width: screenWidth - 32, borderRadius: 16}}
                    resizeMode={'cover'}
                  />
                );
              }}
              keyExtractor={({item, index}) => index}
            />
          </View>
        ) : (
          <View style={{justifyContent: 'center', flex: 1}}>
            <SeeMore numberOfLines={10} linkStyle={styles.textContentFeed}>
              {`${message}`}
            </SeeMore>

            <View style={styles.pollOptionsContainer}>
              {polls.map((item, index) => {
                /*
                  TODO : Count percentage
                */

                return (
                  <PollOptions
                    item={item}
                    index={index}
                    total={totalPollCount}
                  />
                );
              })}
            </View>

            <View style={styles.totalVotesContainer}>
              <Text
                style={styles.totalpolltext}>{`${totalPollCount} votes `}</Text>
              <View
                style={{
                  width: 4,
                  height: 4,
                  borderRadius: 4,
                  alignSelf: 'center',
                  backgroundColor: colors.blackgrey,
                }}
              />
              <Text style={styles.totalpolltext}>
                {' 24 hours 34 minutes left'}
              </Text>
            </View>
          </View>
        )
      ) : null}
    </TouchableOpacity>
  );
};

export default ContentPoll;

const styles = StyleSheet.create({
  rowSpaceBeetwen: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  rowCenter: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  containerFeedProfile: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginLeft: 13,
  },

  feedUsername: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 14,
    color: colors.black,
  },
  containerFeedText: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 5,
  },
  feedDate: {
    fontFamily: fonts.inter[400],
    fontSize: 12,
    color: colors.black,
    lineHeight: 18,
  },
  point: {
    width: 4,
    height: 4,
    borderRadius: 4,
    backgroundColor: colors.gray,
    marginLeft: 8,
    marginRight: 8,
  },
  contentFeed: {
    flex: 1,
    marginTop: 12,
  },
  textContentFeed: {
    fontFamily: fonts.inter[400],
    fontSize: 24,
    lineHeight: 24,
    color: colors.black,
  },
  textComment: {
    fontFamily: fonts.inter[400],
    fontSize: 12,
    lineHeight: 18,
    color: colors.gray,
  },
  usernameComment: {
    fontFamily: fonts.inter[500],
    fontWeight: '900',
    fontSize: 12,
    lineHeight: 24,
    color: colors.black,
  },
  usernameTextComment: {
    fontFamily: fonts.inter[500],
    fontSize: 12,
    lineHeight: 24,
    color: colors.gray,
  },
  item: {
    width: screenWidth - 20,
    height: screenWidth - 20,
    marginTop: 10,
    marginLeft: -20,
    backgroundColor: 'pink',
  },
  imageContainer: {
    flex: 1,
    marginBottom: Platform.select({ios: 0, android: 1}), // Prevent a random Android rendering issue
    backgroundColor: 'white',
    borderRadius: 8,
  },
  image: {
    ...StyleSheet.absoluteFillObject,
    flex: 1,
    aspectRatio: 1.5,
    resizeMode: 'cover',
  },
  imageAnonimity: {
    marginRight: 8,
    width: 32,
    height: 32,
  },
  pollOptionsContainer: {
    // borderColor : 'red',
    // borderWidth : 4,
    width: '100%',
    padding: 0,
    marginTop: 16,
    marginBottom: 8,
  },
  pollOptionItemContainer: {
    // borderColor : colors.holytosca,
    backgroundColor: colors.lightgrey,
    // borderWidth : 1.25,
    marginBottom: 8,
    borderRadius: 8,
    display: 'flex',
    flexDirection: 'row',
  },
  pollOptionItemContainerActive: {
    // borderColor : colors.holytosca,
    backgroundColor: colors.holytosca30percent,
    // borderWidth : 1.25,
    marginBottom: 8,
    borderRadius: 8,
    display: 'flex',
    flexDirection: 'row',
  },
  pollOptionTextContainer: {
    display: 'flex',
    flexDirection: 'row',
    width: '100%',
    paddingVertical: 16,
    paddingHorizontal: 12,
  },
  pollOptionItemText: {
    flex: 1,
    color: colors.black,
    fontFamily: fonts.inter[400],
    // backgroundColor : 'red'
  },
  pollOptionItemPercentage: {
    // backgroundColor : 'red'
  },
  percentageBar: (percent) => {
    if (!percent) {
      percent = 0;
    }
    if (percent > 100) {
      percent = 100;
    }

    return {
      width: `${percent}%`,
      height: '100%',
      position: 'absolute',
      top: 0,
      left: 0,
      backgroundColor: colors.bondi_blue,
    };
  },
  totalpolltext: {
    fontFamily: fonts.inter[400],
    fontSize: 12,
    lineHeight: 16,
    color: colors.blackgrey,
  },
  pollRadioButton: {
    width: 12,
    height: 12,
    alignSelf: 'center',
    borderRadius: 6,
    borderColor: colors.black,
    borderWidth: 1,
    marginEnd: 12,
  },

  pollRadioButtonActive: {
    width: 12,
    height: 12,
    alignSelf: 'center',
    borderRadius: 6,
    backgroundColor: colors.holytosca,
    marginEnd: 12,
  },
  totalVotesContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
});
