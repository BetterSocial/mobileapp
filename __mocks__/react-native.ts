import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN