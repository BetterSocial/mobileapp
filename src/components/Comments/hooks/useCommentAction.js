import React from 'react';
import {Alert} from 'react-native';
import {useNavigation, useRoute} from '@react-navigation/core';
import {Context} from '../../../context';
import StringConstant from '../../../utils/string/StringConstant';
import {deleteComment} from '../../../service/comment';

const useCommentAction = () => {
  const [profile] = React.useContext(Context).profile;
  const {params} = useRoute();
  const {goBack} = useNavigation();
  const showAlertDelete = (item, backToPrevPage, callback) => {
    if (profile.myProfile?.user_id === item?.user?.id || item?.is_you) {
      Alert.alert(null, StringConstant.feedDeleteCommentConfirmation, [
        {
          text: 'No, cancel',
          style: 'cancel'
        },
        {
          text: 'Yes',
          style: 'destructive',
          onPress: () => onDeleteCommentClicked(item, backToPrevPage, callback)
        }
      ]);
    }
  };
  const onDeleteCommentClicked = async (item, backToPrevPage, callback) => {
    try {
      const response = await deleteComment(item?.id);

      if (response?.success) {
        if (params.getComment && typeof params.getComment === 'function') {
          params.getComment();
        }
        if (backToPrevPage) {
          goBack();
        }
        if (callback && typeof callback === 'function') {
          console.log('masuk callback');
          callback();
        }
      }
    } catch (e) {
      console.log(e, 'eman');
    }
  };

  return {
    showAlertDelete
  };
};

export default useCommentAction;
