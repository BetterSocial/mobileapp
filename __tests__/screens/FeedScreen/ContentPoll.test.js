import * as React from 'react';
import moment from 'moment';
import {cleanup, fireEvent, render} from '@testing-library/react-native';

import * as usePollMultiple from '../../../src/components/PollOptionsMultipleChoice/hooks/usePollOptionMultiple';
import ContentPoll from '../../../src/screens/FeedScreen/ContentPoll';
import PollOptinComponent from '../../../src/components/PollOptionsMultipleChoice';
import {getPollTime} from '../../../src/utils/string/StringUtils';

describe('Content poll should same as snapshot', () => {
  const item = {
    post_type: 1,
    message: 'halo',
    images_url: '',
    pollOptions: [
      {
        counter: '0',
        createdAt: '2022-11-28T01:24:25.000Z',
        option: 'bahaya123',
        polling_id: 'dee1ff13-9e1e-46b2-8ffc-46afdd38acca',
        polling_option_id: 'c4d16d91-89f5-423c-acb5-95a1b91ec53b',
        updatedAt: '2022-11-28T01:24:25.000Z'
      },
      {
        counter: '1',
        createdAt: '2022-11-28T01:24:25.000Z',
        option: 'bahaya1',
        polling_id: 'dee1ff13-9e1e-46b2-8ffc-46afdd38acca',
        polling_option_id: '3bc4ec5e-ac8f-4704-8256-1018edd9224c',
        updatedAt: '2022-11-28T01:25:15.000Z'
      }
    ],
    polls_expired_at: '2022-11-29T01:24:25.170Z',
    multiplechoice: true,
    isalreadypolling: true,
    voteCount: 2,
    topics: ['poll']
  };

  const itemNotMultiple = {
    post_type: 1,
    message: 'halo',
    images_url: '',
    pollOptions: [
      {
        counter: '0',
        createdAt: '2022-11-28T01:24:25.000Z',
        option: 'bahaya123',
        polling_id: 'dee1ff13-9e1e-46b2-8ffc-46afdd38acca',
        polling_option_id: 'c4d16d91-89f5-423c-acb5-95a1b91ec53b',
        updatedAt: '2022-11-28T01:24:25.000Z'
      },
      {
        counter: '1',
        createdAt: '2022-11-28T01:24:25.000Z',
        option: 'bahaya1',
        polling_id: 'dee1ff13-9e1e-46b2-8ffc-46afdd38acca',
        polling_option_id: '3bc4ec5e-ac8f-4704-8256-1018edd9224c',
        updatedAt: '2022-11-28T01:25:15.000Z'
      }
    ],
    polls_expired_at: '2022-11-29T01:24:25.170Z',
    multiplechoice: false,
    isalreadypolling: false,
    voteCount: 2,
    topics: ['poll']
  };

  afterEach(cleanup);

  it('should same as snapshot', () => {
    const onnewpollfetched = jest.fn();
    const tree = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={item.polls_expired_at}
        voteCount={item.voteCount}
        isalreadypolling={item.isalreadypolling}
        polls={item.pollOptions}
        currentMoment={moment('2023-04-10T01:24:25.000Z')}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('poll multiplechoice component should be show', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByTestId} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={item.polls_expired_at}
        voteCount={item.voteCount}
        isalreadypolling={item.isalreadypolling}
        polls={item.pollOptions}
      />
    );
    expect(getAllByTestId('multiple')).toHaveLength(2);
  });

  it('poll not multiplechoice component should be show', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByTestId} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={false}
        pollexpiredat={item.polls_expired_at}
        voteCount={item.voteCount}
        isalreadypolling={item.isalreadypolling}
        polls={item.pollOptions}
      />
    );
    expect(getAllByTestId('option')).toHaveLength(2);
  });

  it('poll date should be existed', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByText} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={false}
        pollexpiredat={item.polls_expired_at}
        voteCount={item.voteCount}
        isalreadypolling={item.isalreadypolling}
        polls={item.pollOptions}
      />
    );
    const polltime = getPollTime(item.polls_expired_at);
    expect(getAllByText(polltime)).toHaveLength(1);
  });

  it('total vote text should be existed', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByText} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={false}
        pollexpiredat={'2022-11-29T01:24:25.170Z'}
        voteCount={item.voteCount}
        isalreadypolling={false}
        polls={item.pollOptions}
      />
    );
    expect(getAllByText(`${item.voteCount} votes`)).toHaveLength(1);
  });

  it('renderPercentageBar should run correctly', () => {
    const {getAllByTestId} = render(
      <PollOptinComponent
        mypoll={[]}
        item={item.pollOptions[0]}
        selectedindex={'1'}
        multiplechoice={false}
        isexpired={true}
        isalreadypolling={true}
      />
    );
    expect(getAllByTestId('isExpired')).toHaveLength(1);
    const {getAllByTestId: getAlreadyPolling} = render(
      <PollOptinComponent
        mypoll={[]}
        item={item.pollOptions[0]}
        selectedindex={'1'}
        multiplechoice={false}
        isexpired={false}
        isalreadypolling={true}
      />
    );
    expect(getAlreadyPolling('isAlreadyPolling')).toHaveLength(1);
  });

  it('renderPollBadge should run correctly', () => {
    const {getAllByTestId} = render(
      <PollOptinComponent
        maxpolls={['c4d16d91-89f5-423c-acb5-95a1b91ec53b']}
        mypoll={[]}
        item={item.pollOptions[0]}
        selectedindex={'1'}
        multiplechoice={false}
        isexpired={true}
        isalreadypolling={true}
      />
    );
    expect(getAllByTestId('pollWinner')).toHaveLength(1);
    const {getAllByTestId: getPollNotEndedAndIsMine} = render(
      <PollOptinComponent
        maxpolls={['c4d16d91-89f5-423c-acb5-95a1b91ec53b']}
        mypoll={[{polling_option_id: 'c4d16d91-89f5-423c-acb5-95a1b91ec53b'}]}
        item={item.pollOptions[0]}
        selectedindex={'1'}
        multiplechoice={false}
        isexpired={false}
        isalreadypolling={true}
      />
    );
    expect(getPollNotEndedAndIsMine('isPollNotEndedAndIsMine')).toHaveLength(1);
    const {getAllByTestId: getNoPoll} = render(
      <PollOptinComponent
        maxpolls={['c4d16d91-89f5-423c-acb5-95a1b91ec53a']}
        mypoll={[{polling_option_id: 'c4d16d91-89f5-423c-acb5-95a1b91ec53a'}]}
        item={item.pollOptions[0]}
        selectedindex={'1'}
        multiplechoice={false}
        isexpired={false}
        isalreadypolling={true}
      />
    );
    expect(getNoPoll('nonePoll')).toHaveLength(1);
  });

  it('optionPercentage should occur', async () => {
    const onnewpollfetched = jest.fn();
    const {getAllByTestId} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={item.polls_expired_at}
        voteCount={item.voteCount}
        isalreadypolling={item.isalreadypolling}
        polls={item.pollOptions}
      />
    );
    const mockOnOptionsClicked = jest.fn();
    jest.spyOn(usePollMultiple, 'usePollOptionMultiple').mockImplementation(() => ({
      onOptionsClicked: mockOnOptionsClicked
    }));
    expect(getAllByTestId('optionPercentage')).toHaveLength(2);
    // expect(getAllByText())
  });

  it('checkbolx poll should occured', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByTestId} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={moment().add(2, 'day')}
        voteCount={item.voteCount}
        isalreadypolling={false}
        polls={item.pollOptions}
      />
    );
    expect(getAllByTestId('checkbox')).toHaveLength(2);
    const mockOnOptionsClicked = jest.fn();
    jest.spyOn(usePollMultiple, 'usePollOptionMultiple').mockImplementation(() => ({
      onOptionsClicked: mockOnOptionsClicked
    }));
    fireEvent(getAllByTestId('checkbox')[0], 'change');
  });

  it('option text should be there', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByText} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={moment().add(2, 'day')}
        voteCount={item.voteCount}
        isalreadypolling={false}
        polls={item.pollOptions}
      />
    );
    expect(getAllByText(item.pollOptions[0].option)).toHaveLength(1);
    expect(getAllByText(item.pollOptions[1].option)).toHaveLength(1);
  });

  it('optionPercentage should count correctly', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByText} = render(
      <ContentPoll
        item={item}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={item.multiplechoice}
        pollexpiredat={moment().subtract(2, 'day')}
        voteCount={item.voteCount}
        isalreadypolling={true}
        polls={item.pollOptions}
      />
    );
    expect(getAllByText('0%')).toHaveLength(1);
    expect(getAllByText('50%')).toHaveLength(1);
  });
  it('isExpiredPollOption should have length', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByTestId} = render(
      <ContentPoll
        item={itemNotMultiple}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={itemNotMultiple.multiplechoice}
        pollexpiredat={moment().subtract(2, 'day')}
        voteCount={itemNotMultiple.voteCount}
        isalreadypolling={true}
        polls={itemNotMultiple.pollOptions}
      />
    );
    expect(getAllByTestId('isExpiredPollOption')).toHaveLength(2);
  });

  it('expired date text should correct', () => {
    const onnewpollfetched = jest.fn();
    const {getAllByText} = render(
      <ContentPoll
        item={itemNotMultiple}
        onnewpollfetched={onnewpollfetched}
        multiplechoice={itemNotMultiple.multiplechoice}
        pollexpiredat={moment().subtract(2, 'day')}
        voteCount={itemNotMultiple.voteCount}
        isalreadypolling={true}
        polls={itemNotMultiple.pollOptions}
      />
    );
    expect(getAllByText('Poll closed 2d ago')).toHaveLength(1);
  });
});
