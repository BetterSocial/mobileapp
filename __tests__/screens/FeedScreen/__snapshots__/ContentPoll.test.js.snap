// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content poll should same as snapshot should same as snapshot 1`] = `
<View
  style={
    Object {
      "height": "80%",
      "justifyContent": "flex-start",
      "marginBottom": 0,
      "paddingVertical": 0,
    }
  }
>
  <View
    style={
      Object {
        "padding": 0,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#828282",
          "fontSize": NaN,
          "marginLeft": 2,
        }
      }
    >
      All votes are anonymous - even to the pollâ€™s author!
    </Text>
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#F5F5F5",
            "borderRadius": 8,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
        testID="multiple"
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 6,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
              },
              Object {
                "backgroundColor": "#C4C4C4",
                "width": "0%",
              },
            ]
          }
          testID="isExpired"
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "display": "flex",
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
              "width": "100%",
            }
          }
        >
          <View
            testID="nonePoll"
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#000000",
                "flex": 1,
                "fontFamily": "Inter-Regular",
                "fontSize": NaN,
                "marginBottom": 6,
                "marginLeft": 0,
                "marginTop": 6,
              }
            }
          >
            bahaya123
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": NaN,
              }
            }
            testID="optionPercentage"
          >
            0%
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#F5F5F5",
            "borderRadius": 8,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
        testID="multiple"
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 6,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
              },
              Object {
                "backgroundColor": "#00ADB5",
                "width": "50%",
              },
            ]
          }
          testID="isExpired"
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "display": "flex",
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "paddingVertical": 4,
              "width": "100%",
            }
          }
        >
          <View
            testID="pollWinner"
          >
            <View />
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#000000",
                "flex": 1,
                "fontFamily": "Inter-Regular",
                "fontSize": NaN,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginTop": 0,
              }
            }
          >
            bahaya1
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": NaN,
              }
            }
            testID="optionPercentage"
          >
            50%
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#828282",
          "fontFamily": "Inter-Regular",
          "fontSize": NaN,
          "lineHeight": 16,
        }
      }
    >
      2 votes 
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#828282",
          "borderRadius": 4,
          "height": 4,
          "width": 4,
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#828282",
          "flex": 1,
          "fontFamily": "Inter-Regular",
          "fontSize": NaN,
          "lineHeight": 16,
        }
      }
    >
       Poll closed 100d ago
    </Text>
  </View>
</View>
`;
